sdk/identity/azure-identity/src/main/java/com/azure/identity/AzurePipelinesCredential.java

```diff
@@ -74,7 +74,7 @@ public class AzurePipelinesCredential implements TokenCredential {
                 request.setHeader(HttpHeaderName.AUTHORIZATION, "Bearer " + systemAccessToken);
                 request.setHeader(HttpHeaderName.CONTENT_TYPE, "application/json");
                 try (HttpResponse response = pipeline.sendSync(request, Context.NONE)) {
-                    if (response.getStatusCode() < 200 || response.getStatusCode() >= 400) {
+                    if (response.getStatusCode() != 200) {
                         throw LOGGER.logExceptionAsError(new ClientAuthenticationException("Failed to get the client assertion token "
                             + response.getBodyAsString().block()
```

Comment:
we should get a sync API for this added in the core.
cc: @alzimmermsft

Reply:
good idea. Not a blocker for this beta imo, what do you think?

Reply:
I'd recommend using `response.getBodyAsBinaryData().toString()` as HTTP exceptions should have the response body eagerly read into memory. Also, since this is using `sendSync` is should be safe from blocking on a non-blocking thread.